src/components/Observer/Observer.tsx

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useRef, useState } from 'react';
+import { StyledDiv } from './styles';
+
+export default function Observer({callback}: { callback: Function}) {
+  const [element, setElement] = useState<HTMLDivElement>()
```

Comment:
Please, explain why you need the state to store the ref?

Reply:
I guess it is not necessary in this case, but the concept was to reevaluate the component once the element has changed, but I guess if the element within component would change a reevaluation would happen anyway, so state is not need to retrigger useEffect?